test/e2e/suite/peered_vpc.go

```diff
@@ -423,14 +439,31 @@ type NodeCreate struct {
 
 func CreateNodes(ctx context.Context, test *PeeredVPCTest, nodesToCreate []NodeCreate) {
 	var wg sync.WaitGroup
+
+	test.Logger.Info(fmt.Sprintf("Creating %d nodes. Logging will be paused while nodes are being created and joined to the cluster.", len(nodesToCreate)))
+	test.Logger.Info("Logs will be printed as nodes complete the join process.")
 	for _, entry := range nodesToCreate {
 		wg.Add(1)
 		go func(entry NodeCreate) {
 			defer wg.Done()
 			defer GinkgoRecover()
-			testNode := test.NewTestNode(ctx, entry.InstanceName, entry.NodeName, test.Cluster.KubernetesVersion, entry.OS, entry.Provider, entry.InstanceSize)
+
+			// Create a SwitchWriter to control output from each node
+			// pause output before starting nodes and resume after all nodes are created
+			// this is useful to avoid interleaving of output from different nodes
+			outputControl := e2e.NewSwitchWriter(os.Stdout)
+			outputControl.Pause()
+
+			// Create a new logger that uses our SwitchWriter
+			controlledLogger := e2e.NewPausableLogger(e2e.WithSwitchWriter(outputControl))
+			testNode := test.NewTestNode(ctx, entry.InstanceName, entry.NodeName, test.Cluster.KubernetesVersion, entry.OS, entry.Provider, entry.InstanceSize,
+				WithLogging(controlledLogger, outputControl))
+
 			Expect(testNode.Start(ctx)).To(Succeed(), "node should start successfully")
 			Expect(testNode.Verify(ctx)).To(Succeed(), "node should be fully functional")
+
+			test.Logger.Info(fmt.Sprintf("Node %s created and joined to the cluster.", entry.NodeName))
+			Expect(outputControl.Resume()).To(Succeed(), "should resume output control")
```

Comment:
how do you guarantee only one of these is resumed at once?
i thought bc they run in goroutines, two of them might resume (so print to stdout) at once, which could intermix the messages?